src/lib/y2network/interface_config_builders/br.rb

```diff
@@ -0,0 +1,71 @@
+require "yast"
+require "y2network/config"
+require "y2network/interface_config_builder"
+
+module Y2Network
+  module InterfaceConfigBuilders
+    class Br < InterfaceConfigBuilder
+
+      include Yast::Logger
+
+      def initialize
+        super(type: "br")
+      end
+
+      # @return [Array<Interface>] list of interfaces usable in bridge
+      def select_bridgeable
+        interfaces.all.select { |i| bridgeable?(i) }
+      end
+
+    private
+
+      def interfaces
+        Config.find(:yast).interfaces
+      end
+
+      # Checks whether an interface can be bridged in particular bridge
+      #
+      # @param iface [Interface] an interface to be validated as the bridge slave
+      def bridgeable?(iface)
+        # cannot enslave itself
+        # FIXME: this can happen only bcs we silently use LanItems::Items which
+        # already contains partially configured bridge when adding
+        return false if iface.name == @name
+        return true if !iface.configured
+
+        if interfaces.bond_index[iface.name]
+          log.debug("Excluding (#{iface.name}) - is bonded")
+          return false
+        end
+
+        # the iface is already in another bridge
+        if interfaces.bridge_index[iface.name] && interfaces.bridge_index[iface.name] != @name
+          log.debug("Excluding (#{iface.name}) - already bridged")
+          return false
+        end
+
+        # exclude interfaces of type unusable for bridge
+        case iface.type
+        when "br"
+          log.debug("Excluding (#{iface.name}) - is bridge")
+          return false
+        when "tun", "usb", "wlan"
+          log.debug("Excluding (#{iface.name}) - is #{iface.type}")
+          return false
+        end
```

Comment:
btw I already notice this. at first it can be single branch and also it looks like constant for me. So ideally code like:
```
NON_BRIDGEABLE_TYPES = ["br", "tun", "usb", "wlan"]
...
if NON_BRIDGEABLE_TYPES.include?(iface.type)
  log.debug("Excluding (#{iface.name}) - is #{iface.type}")
  return false
end
```

Reply:
done